pkg/agent/monitoraddnodes.go

```diff
@@ -68,16 +69,34 @@ func (mon *addNodeMonitor) logStatus(status string) {
 	logrus.Infof("Node %s: %s", mon.nodeIPAddress, status)
 }
 
-// MonitorAddNodes waits for the a node to be added to the cluster
+// MonitorAddNodes display the progress of one or more nodes being
+// added to a cluster. ipAddresses is an array of IP addresses to be
+// monitored. clusters is an array of their corresponding initialized Cluster
+// struct used to interact with the assisted-service and k8s APIs.
+func MonitorAddNodes(clusters []*Cluster, ipAddresses []string) {
+	var wg sync.WaitGroup
+
+	for i, ipAddress := range ipAddresses {
+		wg.Add(1)
+
+		go MonitorSingleNode(clusters[i], ipAddress, &wg)
+	}
+
+	wg.Wait()
+}
+
+// MonitorSingleNode waits for the a node to be added to the cluster
 // and reports its status until it becomes Ready.
-func MonitorAddNodes(cluster *Cluster, nodeIPAddress string) error {
+func MonitorSingleNode(cluster *Cluster, nodeIPAddress string, wg *sync.WaitGroup) {
+	defer wg.Done()
 	timeout := 90 * time.Minute
 	waitContext, cancel := context.WithTimeout(cluster.Ctx, timeout)
```

Comment:
Wondering if it could be worth having a global context for all the monitored nodes, instead of recreating a new one from scratch for each monitor

Reply:
Make sense. Done.